macOS/LocalPackages/SubscriptionUI/Sources/SubscriptionUI/Preferences/PreferencesSubscriptionModel.swift

```diff
@@ -911,7 +911,14 @@ hasAnyEntitlement: \(hasAnyEntitlement)
 
         if isUserAuthenticated {
             do {
-                let subscription = try await subscriptionManager.getSubscription(cachePolicy: cachePolicy)
+                var subscription = try? await subscriptionManager.getSubscription(cachePolicy: cachePolicy)
+                if subscription == nil {
+                    subscription = try? await subscriptionManager.getSubscription(cachePolicy: .returnCacheDataDontLoad)
+                }
+                guard let subscription else {
+                    throw SubscriptionEndpointServiceError.noData
+                }
```

Comment:
What is the reason for this logic, there was explicit purpose to pass the required `cachePolicy` and use it. If you check how it is used - on initial load it requests cached data, and then follows with attempt to fetch refreshed data. I am not sure about benefit of this embedded logic that overrides the passed `cachePolicy`.

Reply:
I had a second pass at the problem, and I agree. The previous logic was working, but it was not nice.
For now I'm reverting this change and then I'll work on a better solution.